packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -1329,6 +1331,34 @@ export default class NewVm extends BaseComponent {
 
   // DISKS -----------------------------------------------------------------------
 
+  _getDiskSrs = createSelector(
+    () => this.state.state.existingDisks,
+    () => this.state.state.VDIs,
+    (existingDisks, vdis) => {
+      const getSr = _ => _.$SR || _.SR
+      return uniqBy([...existingDisks, ...vdis], getSr).map(getSr)
```

Comment:
Simply construct a `Set`.

Reply:
I used `uniqBy` to avoid duplicate `SR`.

Reply:
In this code, you are merging two array, using `uniqBy` that iterate on this collection and using `map` to get the sr. You can avoid this loops using one loop and storing SRs in a `Set`.